zerver/views/home.py

```diff
@@ -312,11 +312,13 @@ def home_real(request: HttpRequest) -> HttpResponse:
 
     csp_nonce = generate_random_token(48)
     if user_profile is not None:
+        theme_detection = user_profile.theme_detection
         night_mode = user_profile.night_mode
         is_guest = user_profile.is_guest
         is_realm_admin = user_profile.is_realm_admin
         show_webathena = user_profile.realm.webathena_enabled
     else:  # nocoverage
+        theme_detection = False
```

Comment:
I think we want `True` here -- the goal is to record here the defaults to use for a "web public" user with no account, and theme detection seems right there.